src/vivarium_gates_nutrition_optimization/components/intervention.py

```diff
@@ -78,27 +78,34 @@ def on_initialize_simulants(self, pop_data: SimulantData) -> None:
         pop = self.population_view.subview(["maternal_bmi_anemia_category"]).get(
             pop_data.index
         )
-        pop_update = pd.DataFrame(
-            {"intervention": None},
-            index=pop.index,
-        )
-        baseline_ifa = self.randomness.choice(
-            pop.index,
-            choices=[models.IFA_SUPPLEMENTATION, models.NO_TREATMENT],
-            p=[self.ifa_coverage, RESIDUAL_CHOICE],
-            additional_key="baseline_ifa",
-        )
-        low_bmi = pop["maternal_bmi_anemia_category"].isin(
-            [models.LOW_BMI_NON_ANEMIC, models.LOW_BMI_ANEMIC]
-        )
-        coverage = data_values.INTERVENTION_SCENARIO_COVERAGE.loc[self.scenario]
-        pop_update["intervention"] = np.where(
-            low_bmi, coverage["low_bmi"], coverage["normal_bmi"]
-        )
+        if self.scenario == 'ifa':
+            pop_update = pd.DataFrame(
+                {"intervention": 'ifa'},
+                index=pop.index,
+            )
+            self.population_view.update(pop_update)
```

Comment:
nit: you could put the second of these outside of the if/else and then delete this line.